src/routes/helpers/transformers.js

```diff
@@ -0,0 +1,76 @@
+import SchemaHelper from '../../helpers/SchemaHelper'
+import { warning } from '../../utils/logger'
+
+// Used scopes dict
+const scopedContexts = {
+  dc: 'http://purl.org/dc/elements/1.1/',
+  prov: 'https://www.w3.org/TR/prov-o/#',
+  skos: 'http://www.w3.org/2004/02/skos/core#',
+  rdf: 'https://www.w3.org/2000/01/rdf-schema'
+}
+
+/**
+ * Transform document to a JSON-LD structured document
+ * @param {string} type
+ * @param {Object} data
+ * @returns {Object} JSON-LD structured document
+ */
+export const transformJsonLD = (type, data) => {
+  const schemaHelper = new SchemaHelper()
+  const schemaTypes = schemaHelper.getTypeDescription(type).split(',')
+  const prefixes = Object.keys(scopedContexts)
+
+  // Base JSON-LD document
+  const jsonLdData = {
+    '@context': [
+      'https://schema.org/',
+      scopedContexts
+    ],
+    '@type': schemaTypes
+  }
+
+  // Iterate all keys in the data document
+  Object.keys(data).forEach(key => {
+    // Get the property for the current key
+    const property = schemaHelper.findPropertyType(type, key)
+
+    // Use the current value by default
+    let elementValue = data[key]
+
+    // Transform the value when it's a relational property
+    if (elementValue && schemaHelper.isRelationalProperty(property)) {
+      if (Array.isArray(elementValue)) {
+        elementValue = elementValue.map(id => ({
+          '@id': id
+        }))
+      } else if (typeof elementValue === 'string') {
+        elementValue = {
+          '@id': elementValue
+        }
+      }
+    }
+
+    // Iterate over all property scopes
+    property.description.split(',').forEach(uri => {
+      // Find the scope in the predefined scoped context dictionary
+      const prefix = prefixes.find(prefix => uri.indexOf(scopedContexts[prefix]) === 0)
+
+      // Set the property with the namespace
+      if (prefix) {
+        jsonLdData[`${prefix}:${key}`] = elementValue
```

Comment:
there's a problem here that we found where the name of a field in graphql may not be exactly the name of the predicate as defined in the schema. In our case, we are dealing with `"http://www.w3.org/ns/oa#Motivation"`, where `Motivation` has a different meaning to `motivation`, but our field is called `motivation` in the gql schema.
What should we do in this case? One option is to name the field with the correct capitalisation in the schema, but alternatively I made this commit which uses the docstring to name the field, which might be more semantically correct and allow us a bit more flexibility in the naming of fields: https://github.com/MTG/ce-api/commit/9914a00fcb4a03a166fb0be8ae537c98698f8936

Reply:
we realised that we introduced a problem with this commit if there are 2 different fields on an object that have the same prefix in the schema. Only the last one iterated will come out in the final json

Reply:
hmm, what do you suggest? You're saying that the `Motivation` field has a different meaning than the `motivation` field. Does this also mean they shouldn't share the same value?

I like your approach though, it makes changing the JSON fields a lot easier without needing to change the GraphQL schema.

Reply:
The example that @musicog gave is that:
> oa:motivation is a relationship, oa:Motivation is a type

He can explain the specifics of this case better than I can, we can wait for him to comment.
See the examples that we've used in our branch of the CE here:
https://github.com/MTG/ce-api/blob/b34e8df0f52cc034328c7f050ec6183dca79a0ed/src/schema/type/Annotation.graphql#L22-L23
and here
https://github.com/MTG/ce-api/blob/b34e8df0f52cc034328c7f050ec6183dca79a0ed/src/schema/type/Annotation.graphql#L34-L39
I have some more ideas about how we could improve this conversion, I'll make a more specific comment on the PR.

Reply:
For this particular example (motivation / Motivation): sincere apologies, I got it wrong! 
`oa:Motivation` is a type;
`oa:motivatedBy` is a relationship that links an `oa:Annotation` to a `oa:Motivation`.
The authoritative truth for this is always the ontology definition, i.e. https://www.w3.org/ns/oa#

Reply:
96a58bc1b2fa3466825da7be15ccfe1cbecf13ca